tools/check-swagger

```diff
@@ -1,9 +1,40 @@
 var SwaggerParser = require('swagger-parser');
 
-function validateSwagger(file) {
+function check_duplicate_operationids(api) {
+    function iterate_object_keys(object, func) {
+        for(var key in object) {
+            if(object.hasOwnProperty(key)) {
+                func(key);
+            }
+        }
+    }
+
+    var operationids = [];
+
+    iterate_object_keys(api.paths, function(path) {
+        var endpoint = api.paths[path];
+        iterate_object_keys(endpoint, function(key) {
+            var operationid = endpoint[key].operationId;
+            if(operationid !== undefined) {
+                if(operationids.indexOf(operationid) >= 0) {
```

Comment:
I was a bit reluctant because the original field that we're looking for in the YAML file is without snake_case, but after giving it a second thought you're right: this is better for style consistency.